rero_ils/modules/items/serializers/csv.py

```diff
@@ -91,15 +95,20 @@ def preprocess_search_hit(self, pid, record_hit, links_factory=None,
         document = search_document_by_pid(record['document']['pid'])
         record['document_title'] = title_format_text_head(document.title,
                                                           with_subtitle=True)
+
+        # process contributions
         creator = []
-        for contribution in document.contribution:
-            if any(role in contribution.role for role in role_filter):
-                try:
-                    creator.append(contribution['agent']['preferred_name'])
-                except KeyError:
-                    creator.append(
-                        contribution['agent']['authorized_access_point_en']
-                    )
+        if 'contribution' in document:
+            for contribution in document.contribution:
```

Comment:
You can remove the previous `if` by:
 `for contribution in document.get('contribution', []):` or similar

Reply:
`Get` is not working in this case, because `document` is a search result of type `elasticsearch_dsl.response.hit.Hit` without `get` function. We could use `to_dict()` from the `Hit` class in the `search_document_by_pid` function, but then we have to change the code everywhere where it is used.